IQSS/dataverse-frontend
# Feature/420 Select Advanced (single and multiple selection with search)

**What this PR does / why we need it**:
Adds a new Select component to the design system and integrates it into the EditDatasetMetadata form for multiple controlled vocabularies and for non-multiple controlled vocabularies when they have more than 10 options.

Removes `SelectMultiple` component, this `SelectAdvanced` component behaves as single or multiple selection.

**Which issue(s) this PR closes**:

Closes #420 

**Special notes for your reviewer**:


**Suggestions on how to test this**:
1. Execute `npm i`.
2. Navigate with `cd packages/design-system && npm run build`.
3. Execute `npm run storybook`
4. Test the SelectAdvanced component on Storybook on all of its modes:
- Single
- Multiple
- Single With Default Value
- Multiple With Default Value
- Single not Searchable
- Multiple not Seachable
- Invalid
- Disabled
- With Different Select Word (this only for single mode)
5. Also check the create dataset or edit dataset form and find a vocabulary not multiple field that has more than 10 options, instead of showing a native select it should show this `SelectAdvanced` component in single selection mode.


**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
Yes.

**_SelectAdvanced on single mode with search._**
<img width="570" alt="Screenshot 2024-07-01 at 10 25 16" src="https://github.com/IQSS/dataverse-frontend/assets/64560524/f977bbae-1fd6-46e4-bd64-341356ede10d">

**_SelectAdvanced on single mode without search._**
<img width="568" alt="Screenshot 2024-07-01 at 10 25 26" src="https://github.com/IQSS/dataverse-frontend/assets/64560524/83faf9ce-1fce-4951-932f-4946f15e55f4">


**Is there a release notes update needed for this change?**:
N/A
**Additional documentation**:
N/A
